design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,133 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low (limited growth is still acceptable)
+
+## Metrics for Rook to collect
+Metric names will indicate a hierarchy that can be parsed to add it to  Ceph telemetry collection in
+a more ordered fashion.
+
+For example `rook/version` and `rook/kubernetes/version` should be put into a structure like shown:
+```json
+"rook": {
+  "version": "vx.y.z"
+  "kubernetes": {
+    "version": "vX.Y.Z"
+  }
+}
+```
+
+### Overall metrics
+- `rook/version` - Rook version.
+- `rook/kubernetes/...`
+  - `rook/kubernetes/version` - Kubernetes version.
+    - Ceph already collects os/kernel versions.
+  - `rook/kubernetes/node/count/total` - Number of Kubernetes nodes.
+- `rook/csi/...`
+  - `rook/csi/version` - Ceph CSI version.
+  - `rook/csi/plugin-pod/...` - Info about CSI plugin pods
+    - `rook/csi/plugin-pod/count/...` - Info about how many CSI plugin pods exist and of what types
+      This gives idea of how many k8s nodes are in the k8s cluster that are allowed to use Ceph
+      storage. With different types, telemetry can show if some CSI drivers are ever disabled.
+      - `rook/csi/plugin-pod/count/max` - Max number of pods of any CSI plugin type
+      - `rook/csi/plugin-pod/count/types` - Number of CSI plugin types. This can helpfully show if a
```

Comment:
How is this helpful if we already have the rook version?

Reply:
I think it could be. I think it at least saves us from having to look up or have memorized what version of Rook added new features. I think that seems beneficial.

Reply:
Thinking more since our discussion, I still don't see the benefit. The report itself will show if rbd/ceph/nfs are being used or not. That seems useful enough by itself. Gathering the count of `types` falls in the category of info that tells us there is a disconnect between what rook reports and what ceph gathers, but doesn't help us answer any questions about how rook is being used. If telemetry doesn't tell us how rook is actively being used, I would vote not to collect it.